e2e/peer_ops_test.go

```diff
@@ -95,7 +95,21 @@ func TestAddRemovePeer(t *testing.T) {
 		r.Len(peers, 0)
 	}
 
-	// remove peer: ask g1 to remove g2 as peer
+	devicesDir := testTempDir(t, "devices")
+	r.Nil(prepareLoopDevice(devicesDir+"/gluster_dev1.img", "1", "250M"))
+	_, err = client.DeviceAdd(g2.PeerID(), "/dev/gluster_loop1")
+	r.Nil(err)
+
+	dev, err := client.DeviceList(g2.PeerID(), "/dev/gluster_loop1")
+	r.Nil(err)
+	r.Equal(dev[0].Device, "/dev/gluster_loop1")
+
 	err = client.PeerRemove(g2.PeerID())
```

Comment:
once the peer remove failed, please remove the device and then try removing the peer again